### PR TITLE
Bump the various version-specific numbers to 8.17.

### PR DESCRIPTION
**Pull request by ppedrot**
As per #16085.

Overlays:
- https://github.com/JasonGross/coq-scripts/pull/10
- https://github.com/mit-plv/fiat/pull/72

### CODE REVIEW

**Review by gares — APPROVED** [review]

**Comment by ppedrot** [comment]
Why do we get this weird error?
```
Error: extra files:
theories/Compat/Coq817.v
Makefile.doc:181: recipe for target '_build_vo/default/doc/stdlib//index-list.html' failed
```
The file is part of index-list and exists in theories.

**Review by Zimmi48 — APPROVED** [review]

**Comment by ppedrot** [comment]
I'm not sure how to write an overlay for a submodule whose commit is not present in the main tree yet, so I'm just linking to the relevant patch to coq-scripts in the top post. Since it's backwards compatible once it's merged we can relaunch the CI here.

**Comment by ppedrot** [comment]
Not sure why fiat-parsers is dying now, and the fiat-crypto-ocaml error is even more puzzling. cc @JasonGross

File: dev/ci/user-overlays/16099-ppedrot-bump-8.17.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+overlay fiat_parsers https://github.com/ppedrot/fiat bump-8.17 16099
```
Comment (SkySkimmer) [inline dev/ci/user-overlays/16099-ppedrot-bump-8.17.sh:1]: Please remove since the branch was deleted

**Review by SkySkimmer — APPROVED** [review]

**Comment by ppedrot** [comment]
@JasonGross it seems that the file generated by coq-scripts is buggy. It contains twice the same data, which is the reason it dies.

**Comment by ppedrot** [comment]
Ping @JasonGross. This PR is basically waiting on correct overlays for some subset of {coq-scripts, fiat}, and since I'm not able to write them I'll need your help.

**Comment by JasonGross** [comment]
Sorry for the delay, should be fixed now.  @ppedrot, I have no idea what was wrong; I just re-ran `compatibility/generate-compat-files.sh` and it generated files without the duplication.  I'll restart the CI here

**Comment by ppedrot** [comment]
The CI is now green indeed. @SkySkimmer can you merge swiftly? This should have been done a bit of time ago, right after the branch.

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**